src/rules/no-self-import.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileOverview Prohibits a file from importing itself
+ * @author Gio d'Amelio
+ */
+
+import resolve from 'eslint-module-utils/resolve'
+import isStaticRequire from '../core/staticRequire'
+
+function isImportingSelf(context, node, requireName) {
+  const filePath = context.getFilename()
+  const resolvedPath = resolve(requireName, context)
+  if (filePath === resolvedPath) {
+    context.report({
+        node,
+        // TODO: check in about message, should it print the name of the module itself
+        message: 'Module imports itself',
```

Comment:
That's true - but because resolution is being applied, it might not be immediately apparent which import path is pointing back to the file.